bug: panic when adding a directory

This is on v0.1.0.  

The following commands reproduce, `dir1` should have a few files in it.  There were 3 when I discovered the bug.
```
got add dir1
```
OR
```
got put dir1
```

```
panic: Iterate cannot be called with pending edits [recovered, repanicked]

goroutine 1 [running]:
zombiezen.com/go/sqlite/sqlitex.transaction.func1(0xc0002ace20)
	zombiezen.com/go/sqlite@v1.4.2/sqlitex/savepoint.go:233 +0x1e5
panic({0x10462e0?, 0x133e8d0?})
	runtime/panic.go:783 +0x132
github.com/gotvc/got/src/gotkv.(*Tx).Iterate(0x4600b7?, {0x134c7f8?, 0xc000294c60?}, {{0xc000038540, 0x15, 0x18}, {0xc000038558, 0x15, 0x15}})
	github.com/gotvc/got/src/gotkv/tx.go:99 +0x43a
github.com/gotvc/got/src/gotwc/internal/staging.(*Tx).Iterate(0xc000130690, {0x134c7f8, 0xc000294c60}, {{0xc000038540, 0x15, 0x18}, {0xc000038558, 0x15, 0x15}})
	github.com/gotvc/got/src/gotwc/internal/staging/stage.go:201 +0xa5
github.com/gotvc/got/src/gotwc/internal/staging.(*Tx).CheckConflict(0xc000130690, {0x134c7f8, 0xc000294c60}, {0xc000038528?, 0xc000038528?})
	github.com/gotvc/got/src/gotwc/internal/staging/stage.go:231 +0x2f6
github.com/gotvc/got/src/gotcmd.init.func38.(*WC).Add.1.1({0xc000038528, 0x14}, {{0xc00020e8c0?, 0xc000114710?}, 0x122c858?})
	github.com/gotvc/got/src/gotwc/staging.go:162 +0x8e
go.brendoncarroll.net/state/posixfs.WalkLeaves({0x134c7f8, 0xc000294c60}, {0x13507b0, 0xc000116108}, {0x7ffe5ec568d4, 0x4}, 0xc00004b3e8)
	go.brendoncarroll.net/state@v0.0.0-20241118200920-627c9c196901/posixfs/util.go:146 +0x34c
github.com/gotvc/got/src/gotcmd.init.func38.(*WC).Add.1({0xc000130690, 0x0, {0x7f31503f2a20, 0xc00022a420}, {0x13507b0, 0xc000116108}, 0xc00052a210, 0xc00052a240})
	github.com/gotvc/got/src/gotwc/staging.go:161 +0x16f
github.com/gotvc/got/src/gotwc.(*WC).modifyStaging.func1(0xc0002b42d0, 0xc0002aed00, {0x134c7f8, 0xc000294c60}, 0xc00054f7d8)
	github.com/gotvc/got/src/gotwc/staging.go:79 +0x6dc
github.com/gotvc/got/src/gotwc.(*WC).modifyStaging(0xc0002aed00, {0x134c7f8, 0xc000294c60}, 0xc00004b7d8)
	github.com/gotvc/got/src/gotwc/staging.go:89 +0xa5
github.com/gotvc/got/src/gotwc.(*WC).Add(...)
	github.com/gotvc/got/src/gotwc/staging.go:157
github.com/gotvc/got/src/gotcmd.init.func38({{0x134c7f8, 0xc000294c60}, 0xc000294d20, 0x0, {0x1341d00, 0xc0000ae020}, {0x1341d80, 0xc0000ae028}, {0x1341d80, 0xc0000ae030}, ...})
	github.com/gotvc/got/src/gotcmd/wc_staging.go:33 +0x250
go.brendoncarroll.net/star.Run({0x134c7f8, 0xc000294c60}, {{{0x11efac1, 0x1e}, {0x0, 0x0, 0x0}}, 0x0, {0x1a76070, 0x1, ...}, ...}, ...)
	go.brendoncarroll.net/star@v0.0.0-20251226172902-a1111c13c6b3/context.go:50 +0x37a
github.com/gotvc/got/src/gotcmd.init.NewGroupedDir.func48({{0x134c7f8, 0xc000294c60}, 0xc000294cf0, 0x0, {0x1341d00, 0xc0000ae020}, {0x1341d80, 0xc0000ae028}, {0x1341d80, 0xc0000ae030}, ...})
	go.brendoncarroll.net/star@v0.0.0-20251226172902-a1111c13c6b3/dir.go:98 +0x3be
go.brendoncarroll.net/star.Run({0x134c7f8, 0xc000294c60}, {{{0x0, 0x0}, {0x0, 0x0, 0x0}}, 0x0, {0x0, 0x0, ...}, ...}, ...)
	go.brendoncarroll.net/star@v0.0.0-20251226172902-a1111c13c6b3/context.go:50 +0x37a
go.brendoncarroll.net/star.Main({{{0x0, 0x0}, {0x0, 0x0, 0x0}}, 0x0, {0x0, 0x0, 0x0}, 0xc0002b40f0}, ...)
	go.brendoncarroll.net/star@v0.0.0-20251226172902-a1111c13c6b3/main.go:41 +0x245
github.com/gotvc/got/src/gotcmd.Main()
	github.com/gotvc/got/src/gotcmd/root.go:29 +0x18f
main.main()
	github.com/gotvc/got/cmd/got/main.go:8 +0xf
```

Workaround is to add the files individually.

--- a/src/gotwc/internal/staging/stage.go
+++ b/src/gotwc/internal/staging/stage.go
@@ -198,6 +198,7 @@ func (tx *Tx) Iterate(ctx context.Context, span gotkv.Span) (*Iterator, error) {
 	if err := tx.setup(ctx); err != nil {
 		return nil, err
 	}
+	tx.kvtx.Flush(ctx)
 	it := tx.kvtx.Iterate(ctx, span)
 	return &Iterator{it: it}, nil
 }

